src/Tribe/Feature_Detection.php

```diff
@@ -52,33 +60,78 @@ public function supports_async_process( $force = false ) {
 
 		$cached = get_transient( self::$transient );
 
+		$this->lock_option_name = 'tribe_feature_support_check_lock';
 		if (
 			$force
 			|| false === $cached
 			|| ( is_array( $cached ) && ! isset( $cached['supports_async_process'] ) )
 		) {
-			/*
-			 * Build and dispatch the tester: if it works a transient should be set.
-			 */
-			$tester = new Tribe__Process__Tester();
-			$tester->dispatch();
+			if ( ! $this->has_lock() ) {
+				// Let's avoid race conditions by running two or more checks at the same time.
+				$this->lock();
```

Comment:
Just wondering if the lock mechanism might be slower than the actual task as the lock is done via a write in the DB, so if a instance variable might be a good idea as well combined with the DB

Reply:
Writing to the DB has the precise purpose of avoiding other parallel running processes, running in another PHP thread all-together, from running.
When it comes to different PHP threads then the DB is the only source of truth (excluding the use of proc semaphores that might not be enabled at all).

Reply:
I see so this one might be running in a multi thread env as well? 🎉

Reply:
This code is not made to run in multi-thread but an AJAX call could fire shortly after the requst for a page, or cron, started. On average an admin page requests triggers about ~4 more requests.